src/components/QuestionTopBar.js

```diff
@@ -40,87 +82,79 @@ export default function QuestionTopBar({
       isLastStep() && !allStepsCompleted()
         ? // It's the last step, but not all steps have been completed,
           // find the first step that has been completed
-          steps.findIndex((step, i) => !(i in completed))
-        : currentQuestion + 1;
-    setCurrentQuestion(newActiveStep);
+          questionList
+          ? questionList.findIndex((step, i) => !(i in completed))
+          : currentQuestionIndex + 1
+        : currentQuestionIndex + 1;
+    setCurrentQuestionIndex(newActiveStep);
   };
 
   const handleBack = () => {
-    setCurrentQuestion((prevActiveStep) => prevActiveStep - 1);
+    setCurrentQuestionIndex((prevActiveStep) => prevActiveStep - 1);
   };
 
   const handleStep = (step) => () => {
-    setCurrentQuestion(step);
+    setCurrentQuestionIndex(step);
   };
 
   const handleComplete = () => {
     const newCompleted = completed;
-    newCompleted[currentQuestion] = true;
+    newCompleted[currentQuestionIndex] = true;
     setCompleted(newCompleted);
     handleNext();
   };
 
   const handleReset = () => {
-    setCurrentQuestion(0);
+    setCurrentQuestionIndex(0);
     setCompleted({});
   };
 
   return (
-    <Box sx={{ width: "100%" }}>
-      <Stepper nonLinear alternativeLabel activeStep={currentQuestion}>
-        {steps.map((label, index) => (
-          <Step key={label} completed={completed[index]}>
-            <StepButton color="inherit" onClick={handleStep(index)}>
-              {label}
-            </StepButton>
-          </Step>
-        ))}
-      </Stepper>
-      <div>
+    <div>
+      <Grid container direction={"row"}>
+        <Grid item>
+          <Stepper nonLinear alternativeLabel activeStep={currentQuestionIndex}>
+            {questionList?.map((label, index) => (
+              <Step key={label} completed={completed[index]}>
+                <StepButton color="inherit" onClick={handleStep(index)}>
+                  {`Question ${index + 1}`}
+                </StepButton>
+              </Step>
+            ))}
+            <Step key="plus">
+              <StepLabel
+              icon={<ThemeProvider theme={buttonTheme}>
+              <Fab
+                color="primary"
+                aria-label="add"
+                justify-content="center"
+                style={buttonStyle}
+                onClick={onAddQuestion}
+                align="center"
+              >
+                <AddIcon style={addStyle} />
+              </Fab>
+            </ThemeProvider>}
+              >
+             </StepLabel>
+           </Step>
+          </Stepper>
+        </Grid>
+      </Grid>
+
+      <Grid container direction={"row"} alignItems="left">
         {allStepsCompleted() ? (
```

Comment:
You can reuse `allStepsCompleted ?? components` so you don't need to use `: null`

Reply:
allStepsCompleted returns a boolean, would this expression still work?